apps/glusterfs/volume_entry_allocate.go

```diff
@@ -425,8 +427,58 @@ func (v *VolumeEntry) replaceBrickInVolume(db wdb.DB, executor executors.Executo
 		if err != nil {
 			return err
 		}
-		err = reReadVolEntry.Save(tx)
-		if err != nil {
+
+		updateNodeHostName := true
```

Comment:
I'm afraid this whole block is not correct. We don't need to update the mount info when we replace a brick. The mount host and backup volfile servers should always be *all* nodes in the gluster cluster.

Upon node deletion, when the node is really taken out of the gluster cluster for good, we should update the mount info for each volume...

Reply:
maybe we should do a new patch altogether? sorry.